src/ray/rpc/worker/core_worker_client.h

```diff
@@ -167,8 +169,8 @@ class CoreWorkerClient : public std::enable_shared_from_this<CoreWorkerClient>,
   /// \param[in] address Address of the worker server.
   /// \param[in] port Port of the worker server.
   /// \param[in] client_call_manager The `ClientCallManager` used for managing requests.
-  CoreWorkerClient(const rpc::Address &address, ClientCallManager &client_call_manager)
-      : addr_(address), client_call_manager_(client_call_manager) {
+  CoreWorkerClient(const rpc::Address &address, ClientCallManager &client_call_manager, uint64_t sequence_nr_offset = 0)
+      : addr_(address), client_call_manager_(client_call_manager), sequence_nr_offset_(sequence_nr_offset) {
```

Comment:
So the idea is the client subtracts the sequence numbers so that the server always thinks they start at 0? I think this should work for now but I'm not sure it's the best idea to have these out of sync on client and server because I could envision it causing bugs/confusion in the future. What do you think about instead including the "base sequence number" in the proto message?

Reply:
I don't have a super strong preference, but this seems slightly better to me. You can avoid sending those few bytes unnecessarily, and it would cause all the confusion to leak into the server as well, making the situation even worse IMO.